src/SFA.DAS.EmployerApprenticeshipsService.Application/Queries/GetTransferSenderTransactionDetails/GetTransferTransactionDetailsQueryHandler.cs

```diff
@@ -0,0 +1,66 @@
+﻿using MediatR;
+using SFA.DAS.EAS.Application.Hashing;
+using SFA.DAS.EAS.Domain.Extensions;
+using SFA.DAS.EAS.Domain.Models.Transfers;
+using SFA.DAS.EAS.Infrastructure.Data;
+using SFA.DAS.EAS.Infrastructure.Extensions;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EAS.Application.Queries.GetTransferSenderTransactionDetails
+{
+    public class GetTransferTransactionDetailsQueryHandler : IAsyncRequestHandler<GetTransferTransactionDetailsQuery, GetTransferTransactionDetailsResponse>
+    {
+        private readonly EmployerFinancialDbContext _dbContext;
+        private readonly IPublicHashingService _publicHashingService;
+
+        public GetTransferTransactionDetailsQueryHandler(EmployerFinancialDbContext dbContext,
+            IPublicHashingService publicHashingService)
+        {
+            _dbContext = dbContext;
+            _publicHashingService = publicHashingService;
+        }
+
+        public async Task<GetTransferTransactionDetailsResponse> Handle(GetTransferTransactionDetailsQuery query)
+        {
+            var result = await _dbContext.GetTransfersByTargetAccountId(
+                                    query.AccountId.GetValueOrDefault(),
+                                    query.TargetAccountId,
+                                    query.PeriodEnd);
+
+            var transfers = result as AccountTransfer[] ?? result.ToArray();
+
+            var firstTransfer = transfers.First();
+
+            var senderAccountName = firstTransfer.SenderAccountName;
+            var senderPublicHashedAccountId = _publicHashingService.HashValue(firstTransfer.SenderAccountId);
```

Comment:
I think it makes things easier to use account ID internally rather than hashed ID.